server/webapp/WEB-INF/rails.new/app/controllers/api_v4/admin/pipelines_controller.rb

```diff
@@ -0,0 +1,124 @@
+        end
+      end
+
+      def update
+        result = HttpLocalizedOperationResult.new
+        get_pipeline_from_request
+        pipeline_config_service.updatePipelineConfig(current_user, @pipeline_config_from_request, etag_for(@pipeline_config), result)
+        handle_config_save_or_update_result(result)
+      end
+
+      def destroy
+        result = HttpLocalizedOperationResult.new
+        pipeline_config_service.deletePipelineConfig(current_user, @pipeline_config, result)
+        render_http_operation_result(result)
+      end
+
+      protected
+
+      def get_pipeline_from_request
+        @pipeline_config_from_request ||= PipelineConfig.new.tap do |config|
+          Admin::Pipelines::PipelineConfigRepresenter.new(config).from_hash(params[:pipeline], {go_config: go_config_service.getCurrentConfig()})
+        end
+      end
+
+      def handle_config_save_or_update_result(result, pipeline_name = params[:pipeline_name])
+        if result.isSuccessful
+          load_pipeline(pipeline_name)
+          json = Admin::Pipelines::PipelineConfigRepresenter.new(@pipeline_config).to_hash(url_builder: self)
+          response.etag = [etag_for(@pipeline_config)]
+          render DEFAULT_FORMAT => json
+        else
+          json = Admin::Pipelines::PipelineConfigRepresenter.new(@pipeline_config_from_request).to_hash(url_builder: self)
+          render_http_operation_result(result, {data: json})
+        end
+      end
+
+      def check_for_attempted_pipeline_rename
+        unless CaseInsensitiveString.new(params[:pipeline][:name]) == CaseInsensitiveString.new(params[:pipeline_name])
+          result = HttpLocalizedOperationResult.new
+          result.notAcceptable(LocalizedMessage::string("RENAMING_NOT_ALLOWED", 'pipeline'))
+          render_http_operation_result(result)
+        end
+      end
+
+      def stale_message
+        LocalizedMessage::string("STALE_RESOURCE_CONFIG", 'pipeline', params[:pipeline_name])
+      end
+
+      def etag_for_entity_in_config
+        etag_for(@pipeline_config)
+      end
+
+      def load_pipeline(pipeline_name = params[:pipeline_name])
+        @pipeline_config = pipeline_config_service.getPipelineConfig(pipeline_name)
+        raise RecordNotFound if @pipeline_config.nil?
+      end
+
+      def check_if_pipeline_by_same_name_already_exists
+        if (!pipeline_config_service.getPipelineConfig(params[:pipeline][:name]).nil?)
+          result = HttpLocalizedOperationResult.new
+          result.unprocessableEntity(LocalizedMessage::string("RESOURCE_ALREADY_EXISTS", 'pipeline', params[:pipeline][:name]))
+          render_http_operation_result(result)
+        end
+      end
+
+      def check_if_pipeline_is_defined_remotely
+        if (!@pipeline_config.isLocal())
+          result = HttpLocalizedOperationResult.new
+          pipeline_name = @pipeline_config.name().toString()
```

Comment:
No need for `toString`. You can pass along the case insensitive string to the localized message.